helix-term/src/commands.rs

```diff
@@ -1030,20 +1030,29 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
         .map(|r| text.slice(r.from()..r.to()).to_string())
         .collect();
     let primary = selections.primary();
-    if selections.len() == 1 && primary.to() - primary.from() == 1 {
-        let current_word = movement::move_next_long_word_start(
-            text.slice(..),
-            movement::move_prev_long_word_start(text.slice(..), primary, 1),
-            1,
-        );
+    if primary.len() == 1 {
+        let count = cx.count();
+        let current_word = selections
+            .clone()
+            .transform(|range| {
+                textobject::textobject_word(
+                    text.slice(..),
+                    range,
+                    textobject::TextObject::Inside,
+                    count,
+                    true,
+                )
+            })
+            .primary();
         paths.clear();
         paths.push(
             text.slice(current_word.from()..current_word.to())
                 .to_string(),
         );
     }
     for sel in paths {
-        let p = sel.trim();
+        let surrounding_chars: &[_] = &['\'', '"', '(', ')'];
+        let p = sel.trim().trim_matches(surrounding_chars);
```

Comment:
I don't think we want to trim in all cases: if we have multiple selections or a selection that covers a range with `range.len() > 1`, `gf` should only act on the contents of that selection and shouldn't trim.

I think we should only be trimming in the `if selection.len() == 1 && primary.len() == 1` branch